crates/generator/src/syscalls/mod.rs

```diff
@@ -347,26 +350,66 @@ impl<'a, S: State, C: ChainStore, Mac: SupportMachine> Syscalls<Mac> for L2Sysca
                 Ok(true)
             }
             SYS_GET_SCRIPT_HASH_BY_PREFIX => {
+                let script_hash_addr = machine.registers()[A0].to_u64();
                 // fetch prefix script hash
-                let prefix_addr = machine.registers()[A3].to_u64();
-                let prefix_len = machine.registers()[A4].to_u64();
+                let prefix_addr = machine.registers()[A1].to_u64();
+                let prefix_len = machine.registers()[A2].to_u64();
                 // check prefix len
                 if prefix_len != SCRIPT_HASH_PREFIX_LEN {
+                    log::error!("unexpected script hash prefix length: {}", prefix_len);
                     return Err(VMError::Unexpected);
                 }
                 let script_hash_prefix = load_bytes(machine, prefix_addr, prefix_len as usize)?;
-                let script_hash = match self
-                    .code_store
-                    .get_script_hash_by_prefix(&script_hash_prefix)
-                {
-                    Some(script_hash) => script_hash,
-                    None => {
-                        machine.set_register(A0, Mac::REG::from_u8(ERROR_NOT_FOUND));
-                        return Ok(true);
+                if let Some(script_hash) = self.get_script_hash_by_prefix(&script_hash_prefix) {
+                    machine
+                        .memory_mut()
+                        .store_bytes(script_hash_addr, script_hash.as_slice())?;
+                    machine.set_register(A0, Mac::REG::from_u8(SUCCESS));
+                } else {
+                    machine.set_register(A0, Mac::REG::from_u8(ERROR_NOT_FOUND));
+                }
+                Ok(true)
+            }
+            SYS_RECOVER_ADDRESS => {
+                // gw_recover_address(msg: Byte32, signature: Bytes, code_hash: Byte32) -> Byte20
```

Comment:
This syscall can be resigned as following to be more generalization:

```suggestion
                // gw_recover_account(msg: Byte32, signature: Bytes, code_hash: Byte32) -> Script
```

Reply:
`code_hash` is in the input argument, and `hash_type` is always `type`, how about just return the `script.args`?